### PR TITLE
Nodes: Add AtomicFunctionNode

### PR DESCRIPTION
**Pull request by cmhhelgeson**
**Description**

Adds the ability to execute atomic functions from within compute shaders. A total swaps indicator has been added to the UI of Compute Bitonic Sort.

image

### CODE REVIEW

**Comment by cmhhelgeson** [comment]
Converting to draft since atomic operations actually need to be able to be assigned to variables.

**Comment by cmhhelgeson** [comment]
The solution I arrived at was just adding a storeNode to each of the atomic functions similar to how storageTexture works, which isn't ideal but at least seems cleaner to me.

File: src/nodes/gpgpu/AtomicFunctionNode.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		const a = this.pointerNode;
+		const b = this.valueNode;
+
+		const params = [];
+
+		params.push( `&${ a.build( builder, inputType ) }` );
+		params.push( b.build( builder, inputType ) );
+
+		const methodSnippet = `${ builder.getMethod( method, type ) }( ${params.join( ', ' )} )`;
+		console.log( this.storeNode );
```
Comment (sunag) [inline src/nodes/gpgpu/AtomicFunctionNode.js:52]: Are you still editing? Or should we remove console.log() in order to merge.

Comment (cmhhelgeson) [reply src/nodes/gpgpu/AtomicFunctionNode.js:52]: Hi, apologies that can be removed

**Pull request merged**